lib/host/aws.js

```diff
@@ -83,32 +95,40 @@ export async function configure(baseUrl) {
     shutdown: {
       enabled: true,
       // Always wait 2 minutes minimum prior to shutting down this node.
-      minimumCycleSeconds: minutes(2),
+      minimumCycleSeconds: minutes(2)
     }
   };
 
-  log('metadata', config);
+  let userdata = await getJsonData(`${baseUrl}/user-data`);
+  let userSuppliedData = userdata.data;
+  let securityToken = userdata.securityToken;
+  let provisionerBaseUrl = userdata.provisionerBaseUrl;
 
-  // query the user data for any instance specific overrides set by the
-  // provisioner.
-  let userdata = await request.get(baseUrl + '/user-data').
-    // Buffer entire response into the .text field of the response.
-    buffer(true).
-    // Issue the request...
-    end();
-
-  if (!userdata.ok || !userdata.text) {
-    log('userdata not available')
-    return config;
-  }
   // parse out overrides from user data
-  log('read userdata', { text: userdata.text });
-  let overrides = JSON.parse(userdata.text);
-  for (var key in overrides) config[key] = overrides[key];
+  log('read userdata', { text: userdata });
+
+  delete userdata.data;
+  delete userdata.securityToken;
+  delete userdata.provisionerBaseUrl;
 
-  log('final config', config);
-  return config;
-};
+  let provisioner = new taskcluster.AwsProvisioner({
+    baseUrl: provisionerBaseUrl
+  });
+  // Retrieve secrets
+  let secrets = await provisioner.getSecret(securityToken);
+
+  let credentials = {
+    taskcluster: secrets.credentials
+  };
+  log('read secrets');
+
+  await provisioner.removeSecret(securityToken);
+
+  // Log config for record of configuration but without secrets
+  log('config', config);
+
+  return Object.assign(config, userdata, userSuppliedData, credentials, secrets.data);
```

Comment:
I definitely like this idea and I'll make the changes.  Awesome suggestion, thank you for making me see the errors of my ways :)